### ISSUE TITLE
Restore functionality with current API

### ISSUE THREAD

**Issue by WildWeazel** [naturenet/naturenet-ios#2]
Matt at ACES wants to demo the app at a conference sometime in March, and needs at least the original features working by then.

**Comment by jasoma** [comment]
So far we know of just these two bugs right?

1. Crashes when attempting to submit an observation
2. Crashes when attempting login

**Comment by WildWeazel** [comment]
Right. Both of which involve a sending a request that we changed. I suspect that submitting a design idea would also cause a crash.

**Comment by jasoma** [comment]
Other confirmed bugs:
1. Crashes when trying to sign up as a new user
2. Submitting a design idea does not crash the app but the idea is not submitted

I'm going to create new issues for each crash but keep this one open until we've fixed all the ones we find.

**Comment by jyxia** [comment]
I looked at the code and new login API. It doesn't surprise me the app crashes when calling login and sign up APIs.  As I pointed out before, the iOS app saves the data locally with the exactly same model as the backend database model. So when a user logs in / signs up, the app expects the account detail information from the API response. Plus, it was a `HTTP GET` request for the login.

**Comment by jyxia** [comment]
Yes, I agree to open new a new issue for each bug/crash.

**Comment by jasoma** [comment]
There is also the problem where the data returned is now very different. Previously you would get the whole `Account` data, now the response looks like this:

```json
{
  "data": {
    "affiliation": null,
    "icon_url": "https://dl.dropboxusercontent.com/u/5104407/nntest/avatar.jpg",
    "id": 3,
    "username": "testuser"
  },
  "status_code": 200,
  "status_txt": "OK"
}
```

So the `Account.parseJSON` also fails because of all the forced unwrapping

**Comment by jyxia** [comment]
I have noticed the new API response. Yes, you are exactly right. I wish `SwiftyJSON ` was around earlier. When I found `Alamofire` and `SwiftyJSON`, I knew, they are the two tools I need for NN iOS. But it was too late.
